libbeat/version/version.go

```diff
@@ -1,3 +1,18 @@
 package version
 
 const defaultBeatVersion = "7.0.0-alpha1"
+
+var (
+	buildTime = "unknown"
+	commit    = "unknown"
+)
+
+// BuildTime exposes the compile-time build time information
+func BuildTime() string {
```

Comment:
I like that but concerned about how to handle parse failure, in case of user error.  At the risk of overkill, what about a `versionTime` type with `func String() string` and `func Parse() (time.Time, error)` methods?

Reply:
I was thinking to only have `version.BuildTime() time.Time`. If the value was not set or was set incorrectly then return `time.Time{}`. For me this would be adequate (I could check for validity if my use case required it with `time.IsZero()`).

But if you want to be more explicit in stating that the build time value is bogus then I'd use `version.BuildTime() (time.Time, error)`.

Reply:
updated as requested, now:

```
# valid time
$ go build -i -ldflags "-X github.com/elastic/beats/libbeat/version.buildTime=2018-01-03T17:30:56Z -X github.com/elastic/beats/libbeat/version.commit=1234"
$ ./libbeat version
mockbeat version 9.9.9 (amd64), libbeat 7.0.0-alpha1 [1234 built 2018-01-03 17:30:56 +0000 UTC]

# invalid time
$ go build -i -ldflags "-X github.com/elastic/beats/libbeat/version.buildTime=XXX -X github.com/elastic/beats/libbeat/version.commit=1234"
$ ./libbeat version
mockbeat version 9.9.9 (amd64), libbeat 7.0.0-alpha1 [1234 built unknown]
```